src/xtgeo/grid3d/_grid_etc1.py

```diff
@@ -290,6 +291,91 @@ def get_heights_above_ffl(
     return htop, hbot, hmid
 
 
+def get_property_between_surfaces(
+    grid: Grid,
+    top: RegularSurface,
+    base: RegularSurface,
+    value: int | float = 1,
+    name: str = "between_surfaces",
+) -> GridProperty:
+    """For a grid, create a grid property with value <value> between two surfaces.
+
+    The value would be zero elsewhere, or if surfaces has inactive nodes.
+    """
+
+    grid._xtgformat2()
+    logger.debug("Creating property between surfaces...")
+
+    xmid, ymid, zmid = _internal.grid3d.grid_cell_centers(
+        grid.ncol,
+        grid.nrow,
+        grid.nlay,
+        grid._coordsv,
+        grid._zcornsv,
+        grid._actnumsv,
+        True,  # asmasked, will give Nan for inactive cells; intentional here
+    )
+
+    top_ = top
+    base_ = base
+    if top.yflip == -1:
+        top_ = top.copy()
+        top_.make_lefthanded()
+        logger.debug("Top surface is right-handed, flipping a copy prior to operation")
+    if base.yflip == -1:
+        base_ = base.copy()
+        base_.make_lefthanded()
+        logger.debug("Base surface is right-handed, flipping a copy prior to operation")
+
+    diff = base_ - top_
+    if diff.values.mean() <= 0:
```

Comment:
How will this work if we have crossing surfaces? Should we instead check at each node instead of the mean value? or should we allow crossing surfaces, and just check that not all top nodes are below the base nodes (which implies they have put the surfaces in incorrect position)? e.g.:


```suggestion
    if (diff.values <= 0).all():
```

Reply:
It will work now in the way that it will only add a cell property when the upper surface is above the lower. If crossing in some parts, it should not give a property. I would think that is most flexible?

Reply:
I am just a bit worried about the check here; it will only trigger if the mean is less than 0. 
For crossing surfaces, this will be a bit unreliable, depending on the depths of the parts of the top surface that are respectively above and below the base surface.

Reply:
OK, will now require that top is <= base on all nodes